daemon/config.go

```diff
@@ -501,12 +501,34 @@ func ValidateConfiguration(config *Config) error {
 	if config.IsValueSet("max-concurrent-downloads") && config.MaxConcurrentDownloads != nil && *config.MaxConcurrentDownloads < 0 {
 		return fmt.Errorf("invalid max concurrent downloads: %d", *config.MaxConcurrentDownloads)
 	}
+	if config.MaxConcurrentDownloads ==nil {
+		maxConcurrentDownloads :=defaultMaxConcurrentDownloads
+		config.MaxConcurrentDownloads = &maxConcurrentDownloads
+	}
+	if *config.MaxConcurrentDownloads < 1 {
+		//maxConcurrentDownloads :=defaultMaxConcurrentDownloads
+		//config.MaxConcurrentDownloads = &maxConcurrentDownloads
+		return fmt.Errorf("invalid max concurrent downloads: %d", *config.MaxConcurrentDownloads)
```

Comment:
Then can you remove the check for `config.IsValueSet("max-concurrent-downloads")` and refactor the code for readability?
Also, please add an automated test that fails without this PR but succeeds with this PR?